configure.go

```diff
@@ -0,0 +1,45 @@
+package discovery
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	kv "gopkg.in/clever/kayvee-go.v2"
+)
+
+// m is a convenience type for using kv.
+type m map[string]interface{}
+
+const (
+	templateVar = "SERVICE_%s_%s_%%s"
+)
+
+func getVar(envVar string) string {
+	envVar = strings.ToUpper(envVar)
+	val := os.Getenv(envVar)
+	if val == "" {
+		panic(kv.FormatLog("discovery-go", kv.Error, "missing env var", m{
+			"var": envVar,
+		}))
+	}
+	return val
+}
+
+// Discover finds the specified URL for a service based off of the service's name and which
+// interface you are accessing. Values are found in environment variables fitting the scheme:
+// SERVICE_{SERVICE NAME}_{INTERFACE NAME}_{PROTO,HOST,PORT}.
+func Discover(service, name string) string {
```

Comment:
I would rename this to better indicate it is returning a url.  I would also add functions for getting the host/port/proto on their own.